modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts

```diff
@@ -0,0 +1,254 @@
+      backupKeyShare,
+      recipientIndex,
+      userGpgKey.publicKey,
+      userGpgKey.privateKey
+    );
+
+    const createBitGoMPCParams: AddKeychainOptions = {
+      keyType: 'tss' as KeyType,
+      source: 'bitgo',
+      keyShares: [
+        {
+          from: 'user',
+          to: 'bitgo',
+          publicShare: userToBitgoShare.publicShare,
+          privateShare: userToBitgoShare.encryptedPrivateShare,
+        },
+        {
+          from: 'backup',
+          to: 'bitgo',
+          publicShare: backupToBitgoShare.publicShare,
+          privateShare: backupToBitgoShare.encryptedPrivateShare,
+        },
+      ],
+      userGPGPublicKey: userGpgKey.publicKey,
+      backupGPGPublicKey: userGpgKey.publicKey,
+      enterprise: enterprise,
+    };
+
+    const keychains = await this.baseCoin.keychains().add(createBitGoMPCParams);
+    return keychains;
+  }
+
+  /** @inheritdoc */
+  async createUserKeychain(
+    userGpgKey: openpgp.SerializedKeyPair<string>,
+    userKeyShare: KeyShare,
+    backupKeyShare: KeyShare,
+    bitgoKeychain: Keychain,
+    passphrase: string,
+    originalPasscodeEncryptionCode?: string
+  ): Promise<Keychain> {
+    const bitgoKeyShares = bitgoKeychain.keyShares;
+    if (!bitgoKeyShares) {
+      throw new Error('Missing BitGo key shares');
+    }
+    if (!bitgoKeychain.commonKeychain) {
+      throw new Error(`Missing common key chain: ${bitgoKeychain.commonKeychain}`);
+    }
+
+    const bitGoToUserShare = bitgoKeyShares.find((keyShare) => keyShare.from === 'bitgo' && keyShare.to === 'user');
+    if (!bitGoToUserShare) {
+      throw new Error('Missing BitGo to User key share');
+    }
+
+    const backupToUserShare = await encryptNShare(backupKeyShare, 1, userGpgKey.publicKey, userGpgKey.privateKey);
+    const encryptedNShares: DecryptableNShare[] = [
+      {
+        nShare: backupToUserShare,
+        recipientPrivateArmor: userGpgKey.privateKey,
+        senderPublicArmor: userGpgKey.publicKey,
+      },
+      {
+        nShare: {
+          i: 1,
+          j: 3,
+          publicShare: bitGoToUserShare.publicShare,
+          encryptedPrivateShare: bitGoToUserShare.privateShare,
+        },
+        recipientPrivateArmor: userGpgKey.privateKey,
+        senderPublicArmor: userGpgKey.publicKey,
+      },
+    ];
+
+    const userCombinedKey = await ECDSAMethods.createCombinedKey(
+      userKeyShare,
+      encryptedNShares,
+      bitgoKeychain.commonKeychain
+    );
+
+    const userKeychainParams: AddKeychainOptions = {
+      source: 'user',
+      keyType: 'tss' as KeyType,
+      commonKeychain: bitgoKeychain.commonKeychain,
+      encryptedPrv: this.bitgo.encrypt({
+        input: JSON.stringify(userCombinedKey.signingMaterial),
+        password: passphrase,
+      }),
+      originalPasscodeEncryptionCode,
+    };
+
+    return await this.baseCoin.keychains().add(userKeychainParams);
+  }
+
+  /** @inheritdoc */
+  async createBackupKeychain(
```

Comment:
createBackup and createUser is almost having same logic right? can we combine them to a single function and resuse, provide all the necessary difference via parameters